app/grandchallenge/reader_studies/views.py

```diff
@@ -1429,3 +1468,326 @@ def form_valid(self, form):
             instance.save()
 
         return HttpResponseRedirect(self.get_success_url())
+
+
+class DisplaySetFilesUpdate(ObjectPermissionRequiredMixin, CreateView):
+    model = RawImageUploadSession
+    template_name = "reader_studies/display_set_files_update.html"
+    permission_required = (
+        f"{ReaderStudy._meta.app_label}.change_{DisplaySet._meta.model_name}"
+    )
+    raise_exception = True
+
+    def get_permission_object(self):
+        return self.display_set
+
+    def get_form_class(self):
+        if self.interface.is_image_kind:
+            return UploadRawImagesForm
+        else:
+            return FileForm
+
+    @cached_property
+    def interface(self):
+        return ComponentInterface.objects.get(pk=self.kwargs["interface_pk"])
+
+    @cached_property
+    def display_set(self):
+        return DisplaySet.objects.get(pk=self.kwargs["pk"])
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context.update(
+            {
+                "display_set": self.kwargs["pk"],
+                "interface": self.kwargs["interface_pk"],
```

Comment:
`interface_pk` still used here and in other places.

Reply:
Yes, was too quick to commit via the web UI, sorry!